kernel/src/snapshot.rs

```diff
@@ -71,9 +72,16 @@ impl LogSegment {
 
     fn read_metadata(&self, engine: &dyn Engine) -> DeltaResult<Option<(Metadata, Protocol)>> {
         let schema = get_log_schema().project(&[PROTOCOL_NAME, METADATA_NAME])?;
+        // filter out log files that do not contain metadata or protocol information
+        use Expression as Expr;
+        let filter = Some(Expr::or(
+            Expr::not(Expr::is_null(Expr::Column("metaData.id".into()))),
+            Expr::not(Expr::is_null(Expr::Column(
+                "protocol.min_reader_version".into(),
+            ))),
+        ));
```

Comment:
nit: Using the Expr::column contructor is a little cleaner:
```suggestion
        let filter = Some(Expr::or(
            Expr::not(Expr::is_null(Expr::column("metaData.id"))),
            Expr::not(Expr::is_null(Expr::column("protocol.min_reader_version"))),
        ));
```